pkg/sysctl/sysctl.go

```diff
@@ -6,177 +6,122 @@ package sysctl
-		if !parameterElemRx.MatchString(elem) {
-			return "", ErrInvalidSysctlParameter(name)
-		}
-	}
-	return filepath.Join(append([]string{GetProcfs(), "sys"}, elems...)...), nil
+	return nil
 }
 
-// SetProcfs sets path for the root's /proc. Calling it after GetProcfs causes
-// panic.
-func SetProcfs(path string) {
-	procFsMU.Lock()
-	defer procFsMU.Unlock()
-	if procFsRead {
-		// do not change the procfs after we have gotten its value from GetProcfs
-		panic("SetProcfs called after GetProcfs")
-	}
-	procFs = path
-}
+// Enable enables the given sysctl parameter.
+func (mgr *Manager) Enable(name string) error {
+	txn := mgr.db.WriteTxn(mgr.settings)
+	defer txn.Commit()
 
-// GetProcfs returns the path set in procFs. Executing SetProcFs after GetProcfs
-// might panic depending. See SetProcfs for more info.
-func GetProcfs() string {
-	procFsMU.Lock()
-	defer procFsMU.Unlock()
-	procFsRead = true
-	return procFs
-}
+	_, _, _ = mgr.settings.Insert(txn, &Setting{
+		Name:   name,
+		Val:    "1",
+		Status: reconciler.StatusPending(),
+	})
 
-func writeSysctl(name string, value string) error {
-	path, err := parameterPath(name)
-	if err != nil {
-		return err
-	}
-	f, err := os.OpenFile(path, os.O_RDWR, 0644)
-	if err != nil {
-		return fmt.Errorf("could not open the sysctl file %s: %s",
-			path, err)
-	}
-	defer f.Close()
-	if _, err := io.WriteString(f, value); err != nil {
-		return fmt.Errorf("could not write to the systctl file %s: %s",
-			path, err)
-	}
 	return nil
 }
 
-// Disable disables the given sysctl parameter.
-func Disable(name string) error {
-	return writeSysctl(name, "0")
-}
+// Write writes the given sysctl parameter.
+func (mgr *Manager) Write(name string, val string) error {
+	txn := mgr.db.WriteTxn(mgr.settings)
+	defer txn.Commit()
 
-// Enable enables the given sysctl parameter.
-func Enable(name string) error {
-	return writeSysctl(name, "1")
-}
+	_, _, _ = mgr.settings.Insert(txn, &Setting{
+		Name:   name,
+		Val:    val,
+		Status: reconciler.StatusPending(),
+	})
 
-// Write writes the given sysctl parameter.
-func Write(name string, val string) error {
-	return writeSysctl(name, val)
+	return nil
 }
 
 // WriteInt writes the given integer type sysctl parameter.
-func WriteInt(name string, val int64) error {
-	return writeSysctl(name, strconv.FormatInt(val, 10))
+func (mgr *Manager) WriteInt(name string, val int64) error {
+	txn := mgr.db.WriteTxn(mgr.settings)
+	defer txn.Commit()
+
+	_, _, _ = mgr.settings.Insert(txn, &Setting{
+		Name:   name,
+		Val:    strconv.FormatInt(val, 10),
+		Status: reconciler.StatusPending(),
+	})
+
+	return nil
 }
 
 // Read reads the given sysctl parameter.
-func Read(name string) (string, error) {
-	path, err := parameterPath(name)
-	if err != nil {
-		return "", err
-	}
-	val, err := os.ReadFile(path)
-	if err != nil {
-		return "", fmt.Errorf("Failed to read %s: %s", path, err)
+func (mgr *Manager) Read(name string) (string, error) {
+	txn := mgr.db.ReadTxn()
```

Comment:
Read directly from proc, or perhaps read from settings first and if nothing there then from proc?